lib/src/query.dart

```diff
@@ -0,0 +1,766 @@
+    return QueryCondition(entityId, propertyId, c);
+  }
+
+  QueryCondition between(double p1, double p2) {
+    return _op(ConditionOp._tween, p1, p2);
+  }
+
+  // TODO determine default tolerance: between (target - tolerance, target + tolerance)
+  QueryCondition equal(double p, {double tolerance = 0.01}) {
+    final absTolerance = tolerance.abs();
+    return between(p - absTolerance, p + absTolerance);
+  }
+
+  QueryCondition greater(double p) {
+    return _op(ConditionOp._gt, p, null);
+  }
+
+  QueryCondition less(double p) {
+    return _op(ConditionOp._lt, p, null);
+  }
+
+  QueryCondition operator < (double p) => less(p);
+  QueryCondition operator > (double p) => greater(p);
+  QueryCondition operator == (double p) => equal(p);
+}
+
+class QueryBooleanProperty extends QueryProperty {
+  QueryBooleanProperty(int entityId, int propertyId) : super(entityId, propertyId);
+
+  static const ConditionType type = ConditionType._bytes;
+
+  // TODO let the programmer decide on the resolution via @Property
+  QueryCondition equal(bool p) {
+    final c  = Condition<int>(ConditionOp._eq, type, (p ? 1 : 0));
+    return QueryCondition(entityId, propertyId, c);
+  }
+
+  QueryCondition operator == (bool p) => equal(p);
+}
+
+class OBXOrderFlag {
+  /// Reverts the order from ascending (default) to descending.
+  static final DESCENDING = 1;
+
+  /// Makes upper case letters (e.g. "Z") be sorted before lower case letters (e.g. "a").
+  /// If not specified, the default is case insensitive for ASCII characters.
+  static final CASE_SENSITIVE = 2;
+
+  /// For scalars only: changes the comparison to unsigned (default is signed).
+  static final UNSIGNED = 4;
+
+  /// null values will be put last.
+  /// If not specified, by default null values will be put first.
+  static final NULLS_LAST = 8;
+
+  /// null values should be treated equal to zero (scalars only).
+  static final NULLS_ZERO = 16;
+}
+
+enum ConditionOp {
+  _null,
+  _not_null,
+  _eq,
+  _not_eq,
+  _string_contains,
+  _strings_contain,
+  _string_starts,
+  _string_ends,
+  _gt,
+  _lt,
+  _in,
+  _not_in,
+  _tween,
+  _all,
+  _any
+}
+
+// TODO determine what is used for 'bool' (in the current implementation)
+enum ConditionType {
+  _string,
+  _int32,
+  _int64,
+  _double,
+  _bytes,
+}
+
+class Condition<DartType> {
+  DartType _value, _value2;
+  List<DartType> _list;
+
+  ConditionOp _op;
+  ConditionType _type;
+
+  Condition(this._op, this._type, this._value, [this._value2 = null]);
+  Condition.fromList(this._op, this._type, this._list);
+
+  int _nullness(Pointer<Void> qbPtr, QueryCondition qc, obx_qb_cond_operator_0_dart_t func) {
+    return func(qbPtr, qc._propertyId);
+  }
+}
+
+class StringCondition extends Condition<String> {
+  List<int> orderFlags;
+  bool _caseSensitive = false, _withEqual = false;
+
+  StringCondition(ConditionOp op, ConditionType type, String value, String value2, int caseSensitive, int descending)
```

Comment:
caseSensitive when used with string conditions is not a `OBXOrderFlag` but a boolean. 

FYI: case-sensitive order flag, i.e. not part of this PR because order isn't, handles only ordering, not conditions (as those are distinct)

Reply:
Oops, I'll separate those.